carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/decorator/ExtendedWebElement.java

```diff
@@ -959,11 +988,25 @@ public boolean isElementWithTextPresent(final String text, long timeout) {
     	final String decryptedText = cryptoTool.decryptByPattern(text, CRYPTO_PATTERN);
 		ExpectedCondition<Boolean> textCondition;
 		if (element != null) {
+			ExpectedCondition<Boolean>  tmpCondition = ExpectedConditions.and(ExpectedConditions.visibilityOf(element));
+			boolean tmpResult = waitUntil(tmpCondition, 0);
+			
+			if (!tmpResult && originalException != null && StaleElementReferenceException.class.equals(originalException.getClass())) {
+				LOGGER.debug("StaleElementReferenceException detected in isElementWithTextPresent!");
+				try {
+					refindElement();
+					textCondition = ExpectedConditions.textToBePresentInElement(element, decryptedText);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "textCondition". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://52.53.207.202/sonarqube/coding_rules#rule_key=squid%3AS1854)